model/src/main/java/org/cbioportal/model/MolecularProfile.java

```diff
@@ -36,6 +36,22 @@ public enum ImportType {
         DISCRETE_LONG;
     }
 
+    public enum CnaDatatype {
+        DISCRETE,
+        DISCRETE_LONG;
+
+        public static CnaDatatype findByName(String name) {
+            CnaDatatype result = null;
+            for (CnaDatatype value : values()) {
+                if (value.name().equalsIgnoreCase(name)) {
+                    result = value;
+                    break;
+                }
+            }
+            return result;
+        }
```

Comment:
Guava has a nice method for this. We don't import guava here, but we do in the service layer. You could replace that method with this code, and delete the logic here:

```java
    private boolean isDiscreteCnaDataType(MolecularProfile molecularProfile) {
        return Enums
            .getIfPresent(MolecularProfile.CnaDatatype.class, molecularProfile.getDatatype().toUpperCase())
            .isPresent();
    }
```

Reply:
Oh, the diff changed, and now this isn't used. Wrong PR?

Reply:
Related to these changes: https://github.com/cBioPortal/cbioportal/pull/9847#issuecomment-1318605217